akka-persistence/src/test/scala/akka/persistence/EventAdapterSpec.scala

```diff
@@ -0,0 +1,217 @@
+  final case class UserDataChanged(countryCode: String, age: Int) extends DomainEvent
+
+  class UserAgeTaggingAdapter extends EventAdapter[JournalModel] {
+    val Adult = Set("adult")
+    val Minor = Set("minor")
+
+    override def toJournal(event: Any): JournalModel = event match {
+      case e @ UserDataChanged(_, age) if age > 18 ⇒ Tagged(e, Adult)
+      case e @ UserDataChanged(_, age)             ⇒ Tagged(e, Minor)
+      case e                                       ⇒ NotTagged(e)
+    }
+    override def fromJournal(event: JournalModel, manifest: String): Any =
+      event.payload
+  }
+
+  class ReplayPassThroughAdapter extends UserAgeTaggingAdapter {
+    override def fromJournal(event: JournalModel, manifest: String): Any =
+      event // don't unpack, just pass through the JournalModel
+  }
+
+  // TODO better split out EventReadAdapter and WriteAdapter ???
+  class PayloadUnwrapper extends EventAdapter[JournalModel] {
+    override def toJournal(event: Any): JournalModel = NotTagged(event)
+    override def fromJournal(event: JournalModel, manifest: String): Any = event.payload
+  }
+
+  class LoggingAdapter(system: ActorSystem) extends EventAdapter[Any] {
+    final val log = Logging(system, getClass)
+    override def toJournal(event: Any): Any = {
+      log.info("On its way to the journal: []: " + event)
+      event
+    }
+    override def fromJournal(event: Any, manifest: String): Any = {
+      log.info("On its way out from the journal: []: " + event)
+      event
+    }
+  }
+
+  case object GetState
+  class PersistAllIncomingActor(name: String, override val journalPluginId: String)
+    extends NamedPersistentActor(name) with PersistentActor {
+
+    var state: List[Any] = Nil
+
+    val persistIncoming: Receive = {
+      case GetState ⇒
+        state.reverse.foreach { sender() ! _ }
+      case in ⇒
+        persist(in) { e ⇒
+          state ::= e
+          sender() ! e
+        }
+    }
+
+    override def receiveRecover = {
+      case RecoveryCompleted ⇒ // ignore
+      case e                 ⇒ state ::= e
+    }
+    override def receiveCommand = persistIncoming
+  }
+
+}
+
+abstract class EventAdapterSpec(journalName: String, journalConfig: Config, adapterConfig: Config)
+  extends PersistenceSpec(journalConfig.withFallback(adapterConfig)) with ImplicitSender {
+  import EventAdapterSpec._
+
+  def this(journalName: String, journalConfig: Config) {
+    this(journalName, journalConfig, ConfigFactory.parseString(
+      s"""
+         |akka.persistence.journal {
+         |
+         |  event-adapters {
+         |    age                 = "${classOf[EventAdapterSpec].getCanonicalName}$$UserAgeTaggingAdapter"
+         |    replay-pass-through = "${classOf[EventAdapterSpec].getCanonicalName}$$ReplayPassThroughAdapter"
+         |    unwrap              = "${classOf[EventAdapterSpec].getCanonicalName}$$PayloadUnwrapper"
+         |  }
```

Comment:
I'm not sure we should complicate it with these two possible locations of `event-adapters`.
It is possible to extract common definitions with typesafe config anyway using syntax like:

    inmem.event-adapters = ${common-event-adapters}